src/daft-execution/src/executor/ray/ray_executor.rs

```diff
@@ -0,0 +1,55 @@
+use std::sync::{Arc, Mutex};
+
+use common_error::DaftResult;
+use daft_plan::ResourceRequest;
+
+use crate::{
+    executor::{
+        resource_manager::{ExecutionResources, ResourceManager},
+        Executor,
+    },
+    task::Task,
+};
+
+use super::ray_partition_ref::RayPartitionRef;
+
+#[derive(Debug)]
+pub struct RayExecutor {
+    resource_manager: Arc<Mutex<ResourceManager>>,
+}
+
+impl RayExecutor {
+    pub fn new(resource_capacity: ExecutionResources) -> Self {
+        let resource_manager = Mutex::new(ResourceManager::new(resource_capacity)).into();
+        Self { resource_manager }
+    }
+}
+
+impl Executor<RayPartitionRef> for RayExecutor {
+    fn can_admit(&self, resource_request: &ResourceRequest) -> bool {
```

Comment:
I get the potential issue of the scheduler misusing the current executor APIs, but just to be clear, the current scheduling protocol is:
1. Get all admissible tasks, i.e. tasks whose resource requests can fit into the currently available resources.
2. Use some scheduling policy to choose the best admissible task to submit next.
3. Submit task - this is where the resource request from the task is subtracted from the currently available resources.

I.e there's no concurrent submission on a set of admissible tasks. The fact that a given task is admissible should never be stale since:
1. We always submit a single task after gathering up all admissible tasks (no batch submission after the admissibility query).
2. The only concurrent access that could happen here is if a task finishes, in which case availability will be **increased**, which can't make admissible tasks inadmissible.

And we obviously don't want to hold the lock from `can_admit` to `submit_task` or prematurely decrement the available resources upon `can_admit`, since we want to get **all** admissible tasks and then choose the "best" one to submit next.